core/evancore.vue.libs/src/components/onpage-navigation/onpage-navigation.ts

```diff
@@ -0,0 +1,173 @@
+import EvanComponent from '../../component';
+
+interface Entry {
+  id: string;
+  label: string;
+  offsetTop?: number;
+  disabled?: boolean;
+}
+
+/**
+ * Component to provide simple scroll to element functionality.
+ * Requires a list if ids and labels. Where the ids should be available somewhere in rendered HTML.
+ *
+ * @usage <evan-onpage-navigation :entries="navEntries" scrollContainerSelector=".dapp-wrapper-content" />
+ * @class OnpageNavigationComponent
+ * @selector evan-onpage-navigation
+ */
+@Component
+class OnpageNavigationComponent extends mixins(EvanComponent) {
+  /**
+   * The selector for a container in which we want to scroll.
+   * Will not be updated after once initialized.
+   */
+  @Prop({
+    type: String,
+    default: 'body',
+  }) scrollContainerSelector: string;
+
+  /**
+   * List of labels and ids sorted by its vertical position.
+   */
+  @Prop({
+    default: [],
+  }) entries: Entry[];
+
+  /**
+   * The id of the currently active item.
+   */
+  activeItem = null;
+
+  /**
+   * Store the reference to scroll container element.
+   */
+  scrollContainer = null;
+
+  runtime = null;
+
+  /**
+   * Check element positions after updated.
+   */
+  @Watch('entries')
+  updatePositions(): void {
+    this.updateElementPositions();
+  }
+
+  /**
+   * Initialize navigation and scroll listener.
+   */
+  async created(): Promise<void> {
+    this.activeItem = this.entries[0].id;
+
+    await this.$nextTick(); // wait for rendered;
+    this.updateElementPositions();
+    this.setScrollContainer();
+
+    this.scrollContainer.addEventListener('scroll', this.scrollHandler);
+  }
+
+  /**
+   * Remove event listener.
+   */
+  beforeDestroy(): void {
+    this.scrollContainer.removeEventListener('scroll', this.scrollHandler);
+  }
+
+  /**
+   * Set first item which has bigger y-offset than the scroll position as active.
+   * Needs to be class method outside throttled function to keep vue this context.
+   *
+   * @param offset
+   */
+  handleScroll(offset: number): void {
+    this.activeItem = this.entries.find((item) => item?.offsetTop > offset).id;
+  }
+
+  /**
+   * Watch if user scrolled and set active item accordingly
+   */
+  scrollHandler = throttle(
+    async ({ target: { scrollTop } }) => {
+      this.activeItem = this.handleScroll(scrollTop);
+    },
+    100,
+    { trailing: true },
+  );
+
+  /**
+   * Smoothly scroll to element with target id.
+   *
+   * @param id
+   */
+  scrollToElement(id: string): void {
+    const element = document.getElementById(id);
+
+    if (!element) {
+      this.runtime = this.getRuntime();
+      this.runtime.logger.log(`Target element with id #${id} is not in DOM`, 'error');
+
+      return;
+    }
+
+    this.activeItem = id;
+    this.scrollContainer.scrollTo({
+      behavior: 'smooth',
+      left: element.offsetLeft,
+      top: element.offsetTop,
+    });
+  }
+
+  /**
+   * get the reference to the scroll container.
+   */
+  setScrollContainer(): void {
```

Comment:
get or set? I would chose a different naming. "set" implies some kind of setter for me.